### PR TITLE
Enables first version of a zoom on emails for the time being

### PR DESCRIPTION
**Pull request by LunarX**
Depends on #890

### CODE REVIEW

**Review by FabianDevel — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/MessageBodyScaleListener.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+import android.view.ScaleGestureDetector
+import android.webkit.WebView
+import android.widget.FrameLayout
+import androidx.recyclerview.widget.RecyclerView
+
+class MessageBodyScaleListener(
+    private val webViewScrollView: RecyclerView,
+    private val messageBodyWebView: WebView,
+    private val directParent: FrameLayout
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/MessageBodyScaleListener.kt:28]:
```suggestion
    private val directParent: FrameLayout,
```

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/MessageBodyTouchListener.kt [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import android.view.ScaleGestureDetector
+import android.view.View
+import android.view.ViewParent
+import android.webkit.WebView
+import kotlin.math.abs
+
+class MessageBodyTouchListener(
+    private val webViewScrollView: ViewParent,
+    private val scaleDetector: ScaleGestureDetector,
+    private val scaledTouchSlop: Int
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/MessageBodyTouchListener.kt:30]:
```suggestion
    private val scaledTouchSlop: Int,
```

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/MessageBodyTouchListener.kt [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+) : View.OnTouchListener {
+
+    private var lastX = 0f
+    private var lastY = 0f
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/MessageBodyTouchListener.kt:34]:
```suggestion
    private var lastX = 0.0f
    private var lastY = 0.0f
```

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadAdapter.kt [hunk 4]
```diff
@@ -190,11 +203,20 @@ class ThreadAdapter(context: Context) : RecyclerView.Adapter<ThreadViewHolder>()
...
     }
 
-    private fun createHtmlForPlainText(text: String): String {
-        Jsoup.parse("").apply {
-            body().appendElement("pre").text(text).attr("style", "word-wrap: break-word; white-space: pre-wrap;")
-            return html()
-        }
+    private fun WebSettings.setupWebViewSettings() {
+        loadWithOverviewMode = true
+        useWideViewPort = true
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadAdapter.kt:208]: See https://github.com/Infomaniak/android-kMail/pull/890#discussion_r1176296763

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadAdapter.kt (hunk 4 above)
Comment (LunarX) [reply app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadAdapter.kt:205]: Actually one of the two lines is gonna be removed in one of the next PRs

**Review by FabianDevel — APPROVED** [review]

**Review by KevinBoulongne — APPROVED** [review]

**Pull request merged**